Documentation/ApiOverview/Http/Index.rst

```diff
@@ -128,93 +141,87 @@ TYPO3 provides a small set of helper methods related to HTTP Requests in the cla
+    instead.
 
 .. _http_utility_response_migration:
 
 Migration
 ~~~~~~~~~
 
-Most of the time, a proper PSR-7 Response
-can be passed back to the call stack (request handler). Unfortunately there
-might still be some places, inside the call stack, where it is not possible to
-directly return a PSR-7 response. In such case, a :php:`PropagateResponseException`
-could be thrown. It will automatically be caught by a PSR-15 middleware and the
-given PSR-7 Response will then directly be returned.
+Most of the time, a proper PSR-7 response can be returned to the call stack
+(request handler). Unfortunately there might still be some places within the
+call stack where it is not possible to return a PSR-7 response directly. In such
+a case a :php:`TYPO3\CMS\Core\Http\PropagateResponseException` could be thrown.
+This is automatically caught by a PSR-15 middleware and the given PSR-7 response
+is then returned directly.
 
-.. code-block:: php
+..  code-block:: php
 
-   // Before
-   HttpUtility::redirect('https://example.org', HttpUtility::HTTP_STATUS_303);
+    // Before
+    HttpUtility::redirect('https://example.org/', HttpUtility::HTTP_STATUS_303);
 
-   // After
+    // After
 
-   // Inject PSR-17 ResponseFactoryInterface
-   public function __construct(ResponseFactoryInterface $responseFactory)
-   {
-      $this->responseFactory = $responseFactory
-   }
+    // use Psr\Http\Message\ResponseFactoryInterface
+    // use TYPO3\CMS\Core\Http\PropagateResponseException
 
-   // Create redirect response
-   $response = $this->responseFactory
-      ->createResponse(303)
-      ->withAddedHeader('location', 'https://example.org')
+    // Inject PSR-17 ResponseFactoryInterface
+    public function __construct(ResponseFactoryInterface $responseFactory)
+    {
+        $this->responseFactory = $responseFactory
+    }
 
-   // Return Response directly
-   return $reponse;
+    // Create redirect response
+    $response = $this->responseFactory
+        ->createResponse(303)
+        ->withAddedHeader('location', 'https://example.org/')
 
-   // or throw PropagateResponseException
-   new PropagateResponseException($response);
+    // Return response directly
+    return $response;
 
-.. note::
+    // Or throw PropagateResponseException
+    new PropagateResponseException($response);
 
-   Throwing exceptions for returning an immediate PSR-7 Response is considered
-   as an intermediate solution only, until it's possible to return PSR-7
-   responses in every relevant place. Therefore, the exception is marked
-   as :php:`@internal` and will most likely vanish again in the future.
+..  note::
+    Throwing exceptions for returning an immediate PSR-7 response is only
+    considered as an intermediate solution until it is possible to return PSR-7
+    responses at any relevant place. Therefore, the exception is marked as
+    :php:`@internal` and will most likely vanish in the future.
 
 HttpUtility::setResponseCode
```

Comment:
The methods which have been removed in v12 can be removed from the Text for main, I would think. However it might be easier to do that in a follow up PR that is then not backported? Do you have it on your list?

Reply:
Yup, have a follow-up in mind to use PHP 8.1 functionality in the examples (like constructor property promotion with readonly).